x-pack/metricbeat/module/azure/data.go

```diff
@@ -0,0 +1,175 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+const noDimension = "none"
+
+// EventsMapping will map metric values to beats events
+func EventsMapping(report mb.ReporterV2, metrics []Metric) error {
+	if len(metrics) == 0 {
+		return errors.New("no metrics found")
+	}
```

Comment:
Not sure if we should consider an error to have no metrics in this generic `EventsMapping`, I think it should be acceptable to have no metrics if there are no resources in the account.